config/src/main/java/no/sysco/middleware/kafka/clients/interceptor/config/ConfigHarvesterInterceptor.java

```diff
@@ -0,0 +1,173 @@
+ *
+ * @author SYSCO Middleware
+ * @param <K>
+ * @param <V>
+ */
+public class ConfigHarvesterInterceptor<K, V> implements ProducerInterceptor<K, V>, ConsumerInterceptor<K, V> {
+
+    /**
+     * configurable property for this interceptor that may be used to override default target topic
+     */
+    public static final String TOPIC_NAME = "config.harvester.topic";
+
+    private static final String DEFAULT_TOPIC_NAME = "__client_configs";
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    private String topicName = DEFAULT_TOPIC_NAME;
+
+    //we need to remove interceptors or we are going into endless interception loop
+    private static final String[] IGNORED_PROPERTIES = {
+        ProducerConfig.INTERCEPTOR_CLASSES_CONFIG,
+        ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG,
+        StreamsConfig.PRODUCER_PREFIX + ProducerConfig.INTERCEPTOR_CLASSES_CONFIG,
+        StreamsConfig.CONSUMER_PREFIX + ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG};
+
+    private static final String[] FILTERED_CONFIG_VALUES = new String[]{
+        "ssl.key.password", "ssl.keystore.password"
+    };
+
+    /**
+     * Callback interceptor function
+     *
+     * @param configs user provided congiguration properties
+     */
+    @Override
+    public void configure(final Map<String, ?> configs) {
+
+        MAPPER.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
+
+        this.checkAndAssignTopicName(configs);
+
+        try {
+            final Properties props = new Properties();
+            configs.forEach((k, v) -> {
+                if (null != v && !Arrays.asList(IGNORED_PROPERTIES).contains(k)) {
+                    props.put(k, v);
+                }
+            });
+
+            configs.forEach((k, v) -> {
+                if (Arrays.asList(FILTERED_CONFIG_VALUES).contains(k)) {
+                    filterHelper(configs, k);
+                }
+            });
+
+            //we need to add own serializers which are not here if we intercept consumer or are not string if we are in 
+            //KafkaStreams app
+            props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
+            props.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");
+
+            Utils.createTopic(this.topicName, 1, (short) 1, props);
+            Producer<String, Object> producer = Utils.getProducer(props);
+            //we send config not props so we have all original values
+            String jsonizedConf = MAPPER.writeValueAsString(configs);
+
+            producer.send(new ProducerRecord<>(this.topicName, props.getProperty("client.id"), jsonizedConf));
+        } catch (JsonProcessingException ex) {
+            Logger.getLogger(ConfigHarvesterInterceptor.class.getName()).log(Level.SEVERE, null, ex);
+            throw new RuntimeException(ex.getMessage(), ex);

```

Comment:
This will break clients if interceptor JSON parsing fails, I don't think this is the right behavior. I would suggest managing JSON exceptions when building a message and contain an exception from producer to don't affect clients, just log exception that interceptor has failed.

Reply:
fixed